Mathlib/Tactic/LinearCombination.lean

```diff
@@ -69,14 +61,15 @@ partial def expandLinearCombo (ty : Expr) (stx : Syntax.Term) : TermElabM Expand
   | `($e₁ + $e₂) => do
     match ← expandLinearCombo ty e₁, ← expandLinearCombo ty e₂ with
     | .const c₁, .const c₂ => .const <$> ``($c₁ + $c₂)
-    | .proof p₁, .const c₂ => .proof <$> ``(pf_add_c $p₁ $c₂)
-    | .const c₁, .proof p₂ => .proof <$> ``(c_add_pf $p₂ $c₁)
     | .proof p₁, .proof p₂ => .proof <$> ``(add_pf $p₁ $p₂)
+    | _ , _ => throwError "'linear_combination' is agnostic to the addition of constants"
```

Comment:
nit: I'd argue that if it were agnostic to them, this would be a warning not an error. Throwing an error doesn't sound agnostic to me!

Reply:
OK, let me know if you have alternative wording that still accomplishes the goal of dropping this functionality!

Reply:
Either
```suggestion
    | _ , _ => throwError "'linear_combination' does not support addition of constants as it would have no effect"
```
or
```suggestion
    | .proof p, .const _ | .const _, .proof p =>
      logWarning "'linear_combination' is agnostic to the addition of constants, so this has no effect"
      pure (.proof p)
```
where the second one allows users to ignore the warning

Reply:
I like the second one and have implemented it.  Let's revisit this if needed when the other planned refactors come.